drainer/translator/kafka.go

```diff
@@ -45,81 +50,51 @@ func TiBinlogToSlaveBinlog(infoGetter TableInfoGetter, schema string, table stri
 				DdlQuery:   tiBinlog.GetDdlQuery(),
 			},
 		}
-	} else {
-		slaveBinlog = &obinlog.Binlog{
-			Type:     obinlog.BinlogType_DML,
-			CommitTs: tiBinlog.GetCommitTs(),
-			DmlData:  new(obinlog.DMLData),
-		}
-
-		for _, mut := range pv.GetMutations() {
-			info, ok := infoGetter.TableByID(mut.GetTableId())
-			if !ok {
-				return nil, errors.Errorf("TableByID empty table id: %d", mut.GetTableId())
-			}
+		return slaveBinlog, nil
+	}
+	slaveBinlog := &obinlog.Binlog{
+		Type:     obinlog.BinlogType_DML,
+		CommitTs: tiBinlog.GetCommitTs(),
+		DmlData:  new(obinlog.DMLData),
+	}
 
-			schema, _, ok = infoGetter.SchemaAndTableName(mut.GetTableId())
-			if !ok {
-				return nil, errors.Errorf("SchemaAndTableName empty table id: %d", mut.GetTableId())
-			}
+	for _, mut := range pv.GetMutations() {
+		info, ok := infoGetter.TableByID(mut.GetTableId())
+		if !ok {
+			return nil, errors.Errorf("TableByID empty table id: %d", mut.GetTableId())
+		}
 
-			iter := newSequenceIterator(&mut)
-			for {
-				mutType, row, err := iter.next()
-				if err != nil {
-					if err == io.EOF {
-						break
-					}
-					return nil, errors.Trace(err)
-				}
+		schema, _, ok = infoGetter.SchemaAndTableName(mut.GetTableId())
+		if !ok {
+			return nil, errors.Errorf("SchemaAndTableName empty table id: %d", mut.GetTableId())
+		}
 
-				table := genTable(schema, info)
-				slaveBinlog.DmlData.Tables = append(slaveBinlog.DmlData.Tables, table)
-				tableMutation := new(obinlog.TableMutation)
-				table.Mutations = append(table.Mutations, tableMutation)
-
-				switch mutType {
-				case pb.MutationType_Insert:
-					tableMutation.Type = obinlog.MutationType_Insert.Enum()
-					tableMutation.Row, err = insertRowToRow(info, row)
-					if err != nil {
-						return nil, err
-					}
-				case pb.MutationType_Update:
-					tableMutation.Type = obinlog.MutationType_Update.Enum()
-					tableMutation.Row, tableMutation.ChangeRow, err = updateRowToRow(info, row)
-					if err != nil {
-						return nil, err
-					}
-				case pb.MutationType_DeleteRow:
-					tableMutation.Type = obinlog.MutationType_Delete.Enum()
-					tableMutation.Row, err = deleteRowToRow(info, row)
-					if err != nil {
-						return nil, err
-					}
-				default:
-					return nil, errors.Errorf("unknown mutation type: %v", mutType)
+		iter := newSequenceIterator(&mut)
+		for {
+			table, err := nextTable(schema, info, iter)
```

Comment:
`nextTable` seems confused, how about put `mutType, row, err := iter.next()` out of this function, and change to a new function name?

Reply:
`nextRow`

Reply:
@WangXiangUSTC @GregoryIan Updated.